### ISSUE TITLE
astropy.table writes int8 column in boolean in FITS output

### ISSUE THREAD

**Issue by rongpu** [astropy/astropy#11963]
### Description

When writing an astropy Table containing a np.int8 column into a FITS file, that column becomes boolean in the output
```
import numpy as np
from astropy.table import Table

t = Table()
t['a'] = np.array([-1, 0, 1, 2], dtype=np.int8)
print(t['a'].dtype)  # int8
t.write('test.fits')

tt = Table.read('test.fits')
print(tt['a'].dtype)  # bool
```

### System Details
Python 3.8.5 (default, Sep  4 2020, 02:22:02) 
[Clang 10.0.0 ]
Numpy 1.20.3
astropy 4.2.1

**Comment by embray** [comment]
Thank you for the report.  This is more specifically a lower-level issue about `astropy.io.fits` than `Table` or the unified-io interface.

The FITS standard does not allow for signed bytes in table columns.  It is simply not one of the supported data types for FITS tables.  Only unsigned bytes are supported.

For some historical reason that I can't recall, however, the code interprets an array of signed bytes as a boolean column.  I'd have to investigate more to see why that is.

I think, at the very least, we could provide a better warning about this.

**Comment by embray** [comment]
In any case, if your goal is to write signed bytes to a FITS table, you will have to change your approach, because it's not something supported by the FITS format, for some reason :/

**Comment by rongpu** [comment]
Thanks @embray. For people that are unaware of this "feature" in the FITS standard, it would be completely unexpected that it silently turns integers into booleans. So, yes, I think it would be great if it prompts a warning or error when this happens.

**Comment by embray** [comment]
See also #11996

**Comment by maxnoe** [comment]
> The FITS standard does not allow for signed bytes in table columns. It is simply not one of the supported data types for FITS tables. Only unsigned bytes are supported.

This is incorrect. The FITS standard does allow storing `int8` using the `BSCALE/BZERO`   header keywords. Actually support for this was just added to `astropy.io.fits` in  #11996.

Table should be able to round trip such data through fits.

**Comment by maxnoe** [comment]
Edit: checking writing unsigned ints through the table interface, they are used for this. So the same change as in #11996 for image extensions could be made for table extensions.

**Comment by mhvk** [comment]
From comments on #16502 (which is a duplicate), for tables two changes would need to be made: `L` (logical) columns should be mapped to numpy `b1` instead of `i1`, and `i1` should be supported using a specific BZERO. The first change is perhaps the most worrying - it might well break things.

The mapping is at https://github.com/astropy/astropy/blob/8c21018c0f0d9c7de38ed4f30845f9ae0f825b94/astropy/io/fits/column.py#L29-L64

Detection of signed integer for an unsigned byte could then follow similar logic to that used for the unsigned integer detection bits, at https://github.com/astropy/astropy/blob/8c21018c0f0d9c7de38ed4f30845f9ae0f825b94/astropy/io/fits/column.py#L1405-L1425
